test/Oscoin/Test/P2P/Handshake.hs

```diff
@@ -16,23 +16,25 @@ import           Hedgehog
 import           Test.Tasty (TestTree, testGroup)
 import           Test.Tasty.Hedgehog (testProperty)
 
-{-# ANN module ("HLint: ignore Reduce duplication" :: String ) #-}
-
 tests :: Dict (IsCrypto c) -> TestTree
 tests d = testGroup "Oscoin.Test.P2P.Handshake"
-    [ testProperty "prop_simple" (prop_simpleHandshake d)
-    , testProperty "prop_secure" (prop_secureHandshake d)
+    [ testProperty "prop_simple"            (prop_simple d)
+    , testProperty "prop_simpleRejectsSelf" (prop_simpleRejectsSelf d)
+    , testProperty "prop_secure"            (prop_secure d)
+    , testProperty "prop_secureRejectsSelf" (prop_secureRejectsSelf d)
     ]
 
 -- | For GHCi use.
 props :: Dict (IsCrypto c) -> IO Bool
 props d = checkParallel $ Group "Oscoin.Test.P2P.Handshake"
-    [ ("prop_simple", prop_simpleHandshake d)
-    , ("test_secure", prop_secureHandshake d)
+    [ ("prop_simple",            prop_simple d)
+    , ("prop_simpleRejectsSelf", prop_simpleRejectsSelf d)
+    , ("prop_secure",            prop_secure d)
+    , ("prop_secureRejectsSelf", prop_secureRejectsSelf d)
     ]
 
-prop_simpleHandshake :: forall c. Dict (IsCrypto c) -> Property
-prop_simpleHandshake Dict = withTests 1 . property $ do
+prop_simple :: forall c. Dict (IsCrypto c) -> Property
+prop_simple Dict = withTests 1 . property $ do
```

Comment:
Orthogonal: now that we have a mock crypto implementation, can we remove the `withTests 1` if the reason it was there in the first place was tests running time?

Reply:
No it's because they use actual sockets, which is dawg slow on our current CI